### PR TITLE
Refactor liqbot ahead of use by capsule

### LINKED ISSUE #141: Refactor liqbot ahead of use by Capsule

### PR DESCRIPTION
**Pull request by zale144**
References #141

### CODE REVIEW

File: config/steering_method.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	switch method {
+	case DiscreteThreeLevelString:
+		return DiscreteThreeLevel, nil
+	case CoinAndBinomialString:
+		return CoinAndBinomial, nil
+	}
+	return NotSet, fmt.Errorf("steering method unknown:%s", method)
+}
+
+// Get returns the underlying string
```
Comment (peterbarrow) [inline config/steering_method.go:32]:
```suggestion
// Get returns the underlying string.
```

File: config/steering_method.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+func (s *SteeringMethod) Get() string {
+	return s.String()
+}
+
+// Get returns the underlying string
```
Comment (peterbarrow) [inline config/steering_method.go:37]:
```suggestion
// Get returns the underlying string.
```

File: data/datarequests.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		return nil, err
+	}
+	if len(response.Accounts) == 0 {
+		s.log.WithFields(log.Fields{
+			"type": typ,
+		}).Debug("zero accounts for party")
+		return num.Zero(), nil
+	}
+	if len(response.Accounts) > 1 {
+		return nil, fmt.Errorf("too many accounts for party: %s", len(response.Accounts))
```
Comment (peterbarrow) [inline data/datarequests.go:57]:
```suggestion
		return nil, fmt.Errorf("too many accounts for party: %d", len(response.Accounts))
```

File: data/streamingdata.go [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+	}
+
+	if err := s.processEvents(req, proc); err != nil {
+		return fmt.Errorf("failed to process market events: %w", err)
+	}
+
+	return nil
+}
+
+// Party related events
```
Comment (peterbarrow) [inline data/streamingdata.go:149]:
```suggestion
// Party related events.
```

**Review by peterbarrow — APPROVED** [review]

File: bot/normal/market.go [hunk 5]
```diff
@@ -153,73 +180,122 @@ func (b *Bot) sendVote(ctx context.Context, pubKeyHex string, proposalId string,
...
+		if i == 0 {
+			price = num.UintChain(price).Mul(num.NewUint(105)).Div(num.NewUint(100)).Get()
+			tif = vega.Order_TIME_IN_FORCE_GTC
+		} else if i == 1 {
+			price = num.UintChain(price).Mul(num.NewUint(95)).Div(num.NewUint(100)).Get()
+			tif = vega.Order_TIME_IN_FORCE_GTC
+		}
+
+		if err := b.submitOrder(ctx,
+			400,
```
[the remark this answers is not in the document]
Comment (zale144) [reply bot/normal/market.go:272]: Yes, there are a few more things in this method that should be configurable. But I would like to tackle it in the next PR. I'm not happy with this method in general, but it did the trick of getting the market out of auction, so it was good at the time.

**Pull request merged**